src/provider/data/data_transaction.cpp

```diff
@@ -0,0 +1,34 @@
+#include "provider/data/data_transaction.h"
+
+TransactionStatus::TransactionStatus(std::string status) :
+    m_status(std::move(status)){}
+
+bool TransactionStatus::operator ==(TransactionStatus const& txStatus)
+{
+    return this->m_status == txStatus.m_status;
+}
+
+bool TransactionStatus::isPending() const
+{
+    return (m_status == "received") || (m_status == "pending") || (m_status == "partially-executed");
```

Comment:
I think `partially-executed` is now obsolete - @miiu96 can confirm. But it's fine either way (I think erdjs handles it as well).

Reply:
also "received" is obsolete
so only "pending" status should still be checked

Reply:
You can remove (m_status == "received") and (m_status == "partially-executed") because proxy will response only with the status `pending`.

Reply:
Removed "received" and "partially-executed".
Thanks for heads up !